drift/core/beamtransfer.py

```diff
@@ -831,190 +831,260 @@ def _generate_svdfiles(self, regen=False, skip_svd_inv=False):
         ## m, performing the SVD, combining the beams and then write out the
         ## results.
 
-        # For each `m` collect all the `m` sections from each frequency file,
-        # and write them into a new `m` file. Use MPI if available.
-        for mi in mpiutil.mpirange(self.telescope.mmax + 1):
+        m_list = np.arange(self.telescope.mmax + 1)
+        if mpiutil.rank0:
+            # For each m, check whether the file exists, if so, whether we
+            # can open it. If these tests all pass, we can skip the file.
+            # Otherwise, we need to generate a new SVD file for that m.
+            for mi in m_list:
+                if os.path.exists(self._svdfile(mi)) and not regen:
+
+                    # File may exist but be un-openable, so we catch such an
```

Comment:
Did this even happen to you that file is un-openable? Otherwise we could scrap that try block.

Reply:
If the hdf5 file is corrupt.

Reply:
right, this should address the issues that Mateus was having, that when the job crashed that there were corrupt files?!